worksheet.rb

```diff
@@ -31,4 +31,136 @@
+    dr_id_w_tot_earned << "Driver: #{driver_id}"
+    dr_id_w_tot_earned << "Total rides: #{nested_ride_info.length}\n\n"
+    big_array << dr_id_w_tot_earned
+  end
+  return big_array
+end
+
+
+def total_each_driver(log)
+  big_array = ["\nDRIVER EARNINGS:\n\n"]
+
+  log.each do |driver_id, nested_ride_info|
+    dr_id_w_tot_earned = []
+    dr_id_w_tot_earned << "Driver: #{driver_id}"
+
+    total_earned = sum(nested_ride_info.map {|column| column[1].to_f})
+    dr_id_w_tot_earned << "Total earned: #{to_dollars(total_earned)}\n\n" #https://stackoverflow.com/questions/11688466/select-all-elements-from-one-column-in-an-array-of-arrays-in-ruby
+    big_array << dr_id_w_tot_earned
+  end
+  return big_array
+end
+
+
+def avg_rating_each_driver(log) #average to be computed & outputted as a float
+big_array = ["\n\nAVERAGE RATINGS:\n\n"]
+
+log.each do |driver_id, nested_ride_info|
+  dr_id_w_tot_earned = []
+  dr_id_w_tot_earned << "Driver: #{driver_id}"
+  average = sum(nested_ride_info.map {|column| column[3].to_f}) / (nested_ride_info.length) #computes average: gets total of ratings, div by no. of rides
+  dr_id_w_tot_earned << "Average rating: #{average}\n\n"
+  big_array << dr_id_w_tot_earned
+end
+return big_array
+end
+
+
+def highest_earners(log)
+  big_array = []
+
+  log.each do |driver_id, nested_ride_info|
+    dr_id_w_tot_earned = []
+    dr_id_w_tot_earned << driver_id
+    dr_id_w_tot_earned << sum(nested_ride_info.map {|column| column[1]})
+    big_array << dr_id_w_tot_earned
+
+  end
+  max_earned = big_array.map(&:last).max
+
+  highest_earner_arr = big_array.each do |dr_id_w_tot_earned, index|
+    dr_id_w_tot_earned[index] == max_earned #picks up any ties :)
+  end #nested array that holds array w paired info for any max earners
+
+  outputs = ["\nHIGHEST EARNER(S):\n\n"]
+
+  highest_earner_arr.each_with_index do |high_earner, index|
+    outputs << "Driver #{high_earner[0]} => #{to_dollars(high_earner[1])}"  #might have to move to_dollars() (not sure how the original values being foat instaed might change the equivalency on line  )
+  end
+  return outputs
+end
+
+
+def highest_rated_drivers(log)
+  big_array = []
+
+  log.each do |driver_id, nested_ride_info|
+    dr_id_w_avg_rating = []
+    dr_id_w_avg_rating << driver_id
+
+    average = sum(nested_ride_info.map {|column| column[3].to_f}) / (nested_ride_info.length)
+    dr_id_w_avg_rating << average
+
+    big_array << dr_id_w_avg_rating
+  end
+
+  max_avg_rating = big_array.map(&:last).max
+
+  highest_rated_arr = big_array.select.with_index do |dr_id_w_avg_rating, index|
+    dr_id_w_avg_rating[1] == max_avg_rating
+  end
+
+  outputs = ["\nHIGHEST RATED DRIVER(S):\n\n"]
+
+  highest_rated_arr.each_with_index do |high_rated, index|
+    outputs << "Driver #{high_rated[0]} => Avg. #{high_rated[1]}\n\n"
+  end
+  return outputs
+end
+
+puts "\n\n---WELCOME TO THE RIDE SHARE LOGGER---\n\n"
+puts rides_each_driver(ride_log)
+puts total_each_driver(ride_log)
```

Comment:
Since these two methods both calculate totals for drivers, it might simplify and speed up your code to combine this functionality into a single method.